tests/test_dependency_runtime_errors.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+from anyio import open_file
+from fastapi import Depends, FastAPI
+from fastapi.testclient import TestClient
+
+app_reraises = FastAPI()
+app_doesnt_reraise = FastAPI()
+
+
+async def get_username_reraises():
+    try:
+        async with await open_file("/path/to/sanchez.txt", "r") as f:
+            yield await f.read()  # pragma: no cover
+    except OSError as ex:
+        raise RuntimeError("File something something, wubba lubba dub dub!") from ex
```

Comment:
```suggestion
        raise RuntimeError("File read error") from ex
```
I think more neutral text would be better